pkg/feature/servicemesh/servicemesh_feature.go

```diff
@@ -0,0 +1,59 @@
+package servicemesh
+
+import (
+	"path"
+	"path/filepath"
+
+	ctrlLog "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
+)
+
+const templatesDir = "templates/servicemesh"
+
+var log = ctrlLog.Log.WithName("features")
+
+func ConfigureServiceMeshFeatures(s *feature.FeaturesInitializer) error {
+	var rootDir = filepath.Join(feature.BaseOutputDir, s.DSCInitializationSpec.ApplicationsNamespace)
+	if err := feature.CopyEmbeddedFiles(templatesDir, rootDir); err != nil {
+		return err
+	}
+
+	serviceMeshSpec := s.ServiceMesh
+
+	smcpCreation, errSmcp := feature.CreateFeature("mesh-control-plane-creation").
+		For(s.DSCInitializationSpec).
+		Manifests(
+			path.Join(rootDir, templatesDir, "/base"),
+		).
+		PreConditions(
+			EnsureServiceMeshOperatorInstalled,
+			feature.CreateNamespace(serviceMeshSpec.ControlPlane.Namespace),
+		).
+		PostConditions(
+			feature.WaitForPodsToBeReady(serviceMeshSpec.ControlPlane.Namespace),
+		).
+		Load()
+	if errSmcp != nil {
+		return errSmcp
```

Comment:
i would assume, in a cluster without servicemesh operator installed yet, this will return errSmcp and not have 
`features	applying manifest	{"feature": "mesh-control-plane-creation", "path": "/tmp/odh-operator/opendatahub/templates/servicemesh/base/smcp.yaml"}` in operator log. 

when  i am testing with live-build, i do not have any of 2 dependent operators installed, and i have both serverless and servicemesh set to Managed, and still gets above log then it stopped to reconcile at 
`ERROR	features	failed to create resource .... `

Reply:
right, now i see. the check is actually only on if certain CRD is in place. so my cluster had these 2 operator installed previously and uninstalled. but the CRD still left in the cluster.

Reply:
I think it is common to not delete CRDs, after a component is uninstalled.

I'm not sure if we should go that deep into checking if the operator pod is actually running. It is going to be a less-reliable check, if the operator can go to any namespace. Personally, I'm happy if the installation stops after timing out checking ServiceMesh pods are missing.